presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -315,8 +315,8 @@ public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Opt
     public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel, Optional<JoinFilterFunction> joinFilterFunction)
     {
         try {
-            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels)
-                    .createPagesHashStrategy(ImmutableList.copyOf(channels), hashChannel);
+            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels, joinFilterFunction.map(f -> f.getClass()))
```

Comment:
Yeah - I passed the JFF class as one of the code versions declared the `joinFilterFunction` field in generated code as that class and not as just `JoinFilterFunction`. But it did not have any impact on performance so I dropped that.

I will replace Optional with boolean flag.